samples/extensions/dynamic_multisample_rasterization/dynamic_multisample_rasterization.cpp

```diff
@@ -0,0 +1,1015 @@
+	render_pass_create_info.pDependencies          = dependencies.data();
+
+	VK_CHECK(vkCreateRenderPass(device->get_handle(), &render_pass_create_info, nullptr, &render_pass));
+}
+
+// Create attachment that will be used in a framebuffer.
+void DynamicMultisampleRasterization::setup_color_attachment()
+{
+	VkImageCreateInfo image_create_info{};
+	image_create_info.sType       = VK_STRUCTURE_TYPE_IMAGE_CREATE_INFO;
+	image_create_info.imageType   = VK_IMAGE_TYPE_2D;
+	image_create_info.format      = render_context->get_format();
+	image_create_info.extent      = {get_render_context().get_surface_extent().width, get_render_context().get_surface_extent().height, 1};
+	image_create_info.mipLevels   = 1;
+	image_create_info.arrayLayers = 1;
+	image_create_info.samples     = sample_count;
+	image_create_info.tiling      = VK_IMAGE_TILING_OPTIMAL;
+	image_create_info.usage       = VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT | VK_IMAGE_USAGE_SAMPLED_BIT | VK_IMAGE_USAGE_INPUT_ATTACHMENT_BIT;
+
+	VK_CHECK(vkCreateImage(device->get_handle(), &image_create_info, nullptr, &color_attachment.image));
+	VkMemoryRequirements memReqs{};
+	vkGetImageMemoryRequirements(device->get_handle(), color_attachment.image, &memReqs);
+
+	VkMemoryAllocateInfo memory_allocation{};
+	memory_allocation.sType           = VK_STRUCTURE_TYPE_MEMORY_ALLOCATE_INFO;
+	memory_allocation.allocationSize  = memReqs.size;
+	memory_allocation.memoryTypeIndex = device->get_memory_type(memReqs.memoryTypeBits, VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT);
+	VK_CHECK(vkAllocateMemory(device->get_handle(), &memory_allocation, nullptr, &color_attachment.mem));
+	VK_CHECK(vkBindImageMemory(device->get_handle(), color_attachment.image, color_attachment.mem, 0));
+
+	VkImageViewCreateInfo image_view_create_info{};
+	image_view_create_info.sType                           = VK_STRUCTURE_TYPE_IMAGE_VIEW_CREATE_INFO;
+	image_view_create_info.viewType                        = VK_IMAGE_VIEW_TYPE_2D;
+	image_view_create_info.image                           = color_attachment.image;
+	image_view_create_info.format                          = render_context->get_format();
+	image_view_create_info.subresourceRange.baseMipLevel   = 0;
+	image_view_create_info.subresourceRange.levelCount     = 1;
+	image_view_create_info.subresourceRange.baseArrayLayer = 0;
+	image_view_create_info.subresourceRange.layerCount     = 1;
+	image_view_create_info.subresourceRange.aspectMask     = VK_IMAGE_ASPECT_COLOR_BIT;
+	// Stencil aspect should only be set on depth + stencil formats (VK_FORMAT_D16_UNORM_S8_UINT..VK_FORMAT_D32_SFLOAT_S8_UINT
+	if (depth_format >= VK_FORMAT_D16_UNORM_S8_UINT)
+	{
+		image_view_create_info.subresourceRange.aspectMask |= VK_IMAGE_ASPECT_STENCIL_BIT;
+	}
+	VK_CHECK(vkCreateImageView(device->get_handle(), &image_view_create_info, nullptr, &color_attachment.view));
+}
+
+void DynamicMultisampleRasterization::setup_depth_stencil()
```

Comment:
`setup_depth_stencil` is called twice, once directly from `ApiVulkanSample::prepare`, and then from `setup_framebuffer`. Thus, all the depth_stencil handles are created twice, but destroyed just once.